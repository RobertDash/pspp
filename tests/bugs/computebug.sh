#!/bin/sh

# This program tests for a bug in the `compute' command

TEMPDIR=/tmp/pspp-tst-$$
TESTFILE=$TEMPDIR/`basename $0`.sps

here=`pwd`;

# ensure that top_srcdir is absolute
cd $top_srcdir; top_srcdir=`pwd`

export STAT_CONFIG_PATH=$top_srcdir/config


cleanup()
{
     rm -rf $TEMPDIR
}


fail()
{
    echo $activity
    echo FAILED
    cleanup;
    exit 1;
}


no_result()
{
    echo $activity
    echo NO RESULT;
    cleanup;
    exit 2;
}

pass()
{
    cleanup;
    exit 0;
}

mkdir -p $TEMPDIR

activity="copy file"
cp $top_srcdir/tests/bugs/computebug.stat $TEMPDIR
if [ $? -ne 0 ] ; then no_result ; fi

activity="chdir"
cd $TEMPDIR
if [ $? -ne 0 ] ; then no_result ; fi


activity="run program"
$SUPERVISOR $here/../src/pspp -o raw-ascii $TEMPDIR/computebug.stat
if [ $? -ne 0 ] ; then no_result ; fi


activity="compare output"
diff -b -B -w $TEMPDIR/pspp.list $top_srcdir/tests/bugs/computebug.out
if [ $? -ne 0 ] ; then fail ; fi

pass;
