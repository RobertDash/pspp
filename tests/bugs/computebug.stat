DATA LIST LIST
 /A (A161)
 B (A3).

BEGIN DATA
abc   def
ghi   jkl
END DATA.

COMPUTE A=upcase(A).
EXECUTE.
LIST.


