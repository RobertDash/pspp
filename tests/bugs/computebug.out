1.1 DATA LIST.  Reading free-form data from INLINE.
+--------+------+
|Variable|Format|
#========#======#
|A       |A161  |
|B       |A3    |
+--------+------+
                                                                                                                                                                A   B
----------------------------------------------------------------------------------------------------------------------------------------------------------------- ---
ABC                                                                                                                                                               def 
GHI                                                                                                                                                               jkl 
